scheduling/fixed_priority_pre_emptive_scheduling.py

```diff
@@ -0,0 +1,106 @@
+# Reference - https://en.wikipedia.org/wiki/Fixed-priority_pre-emptive_scheduling
+# https://en.wikipedia.org/wiki/Scheduling_(computing)
+from __future__ import annotations
+
+
+def calculate_waiting_times(duration_times: list[int]) -> list[int]:
+    """
+    This function calculates the waiting time of some processes that have a
+    specified duration time.
+        Return: The waiting time for each process.
+    >>> calculate_waiting_times([5, 10, 15])
+    [0, 5, 15]
+    >>> calculate_waiting_times([1, 2, 3, 4, 5])
+    [0, 1, 3, 6, 10]
+    >>> calculate_waiting_times([10, 3])
+    [0, 10]
+    """
+    waiting_times = [0] * len(duration_times)
+    for i in range(1, len(duration_times)):
+        waiting_times[i] = duration_times[i - 1] + waiting_times[i - 1]
+    return waiting_times
+
+
+def calculate_turnaround_times(
+    duration_times: list[int], waiting_times: list[int]
+) -> list[int]:
+    """
+    This function calculates the turnaround time of some processes.
+        Return: The time difference between the completion time and the
+                arrival time.
+                Practically waiting_time + duration_time
+    >>> calculate_turnaround_times([5, 10, 15], [0, 5, 15])
+    [5, 15, 30]
+    >>> calculate_turnaround_times([1, 2, 3, 4, 5], [0, 1, 3, 6, 10])
+    [1, 3, 6, 10, 15]
+    >>> calculate_turnaround_times([10, 3], [0, 10])
+    [10, 13]
+    """
+    return [
+        duration_time + waiting_times[i]
+        for i, duration_time in enumerate(duration_times)
+    ]
+
+
+def calculate_average_turnaround_time(turnaround_times: list[int]) -> float:
+    """
+    This function calculates the average of the turnaround times
+        Return: The average of the turnaround times.
+    >>> calculate_average_turnaround_time([0, 5, 16])
+    7.0
+    >>> calculate_average_turnaround_time([1, 5, 8, 12])
+    6.5
+    >>> calculate_average_turnaround_time([10, 24])
+    17.0
+    """
+    return sum(turnaround_times) / len(turnaround_times)
+
+
+def calculate_average_waiting_time(waiting_times: list[int]) -> float:
+    """
+    This function calculates the average of the waiting times
+        Return: The average of the waiting times.
+    >>> calculate_average_waiting_time([0, 5, 16])
+    7.0
+    >>> calculate_average_waiting_time([1, 5, 8, 12])
+    6.5
+    >>> calculate_average_waiting_time([10, 24])
+    17.0
+    """
+    return sum(waiting_times) / len(waiting_times)
+
+
+if __name__ == "__main__":
+    # process id's
+    processes = [1, 2, 3]
+
+    # ensure that we actually have processes
+    if len(processes) == 0:
```

Comment:
Safer if the user passes us None, False, 0, 0.0, etc.
```suggestion
    if not processes:
```